Code/Framework/AzCore/AzCore/std/time.h

```diff
@@ -18,4 +20,133 @@ namespace AZStd
+            int{ currentYmd.year() },
+            unsigned{ currentYmd.month() },
+            unsigned{ currentYmd.day() },
+            static_cast<unsigned>(currentTimeOfDay.hours().count()),
+            static_cast<unsigned>(currentTimeOfDay.minutes().count()),
+            static_cast<unsigned>(currentTimeOfDay.seconds().count()));
+
+        return true;
+    }
+
+    // formats a UTC timestamp with fractional millseconds
+    // Returns true if utcTime is valid
+    template<class Duration>
+    bool GetUTCTimestampMilliseconds(UTCTimestampString& utcTimestamp, const AZStd::chrono::utc_time<Duration>& utcTime)
+    {
+        AZStd::chrono::year_month_day currentYmd;
+        AZStd::chrono::hh_mm_ss<typename AZStd::remove_cvref_t<decltype(utcTime)>::duration> currentTimeOfDay;
+        if (!Internal::GetYearMonthDayWithHoursMinutesSecondsFromUTCTime(currentYmd, currentTimeOfDay, utcTime))
+        {
+            return false;
+        }
+
+        // Print the current utc timestamp with fractional seconds up to milliseconds
+        // using the format specified at https://www.w3.org/TR/NOTE-datetime
+        utcTimestamp = UTCTimestampString::format(
+            "The current UTC time is %.4i-%.2u-%.2uT%.2u:%.2u:%.2u.%.3uZ\n",
+            int{ currentYmd.year() },
+            unsigned{ currentYmd.month() },
+            unsigned{ currentYmd.day() },
+            static_cast<unsigned>(currentTimeOfDay.hours().count()),
+            static_cast<unsigned>(currentTimeOfDay.minutes().count()),
+            static_cast<unsigned>(currentTimeOfDay.seconds().count()),
+            static_cast<unsigned>(currentTimeOfDay.subseconds().count()));
+
+        return true;
+    }
+
+    // formats a UTC timestamp in microseconds
+    // Returns true if utcTime is valid
+    template<class Duration>
+    bool GetUTCTimestampMicroseconds(UTCTimestampString& utcTimestamp, const AZStd::chrono::utc_time<Duration>& utcTime)
+    {
+        AZStd::chrono::year_month_day currentYmd;
+        AZStd::chrono::hh_mm_ss<typename AZStd::remove_cvref_t<decltype(utcTime)>::duration> currentTimeOfDay;
+        if (!Internal::GetYearMonthDayWithHoursMinutesSecondsFromUTCTime(currentYmd, currentTimeOfDay, utcTime))
+        {
+            return false;
+        }
+
+        // Print the current utc timestamp with fractional seconds up to microseconds
+        // using the format specified at https://www.w3.org/TR/NOTE-datetime
+        utcTimestamp = UTCTimestampString::format(
+            "The current UTC time is %.4i-%.2u-%.2uT%.2u:%.2u:%.2u.%.6uZ\n",
+            int{ currentYmd.year() },
+            unsigned{ currentYmd.month() },
+            unsigned{ currentYmd.day() },
+            static_cast<unsigned>(currentTimeOfDay.hours().count()),
+            static_cast<unsigned>(currentTimeOfDay.minutes().count()),
+            static_cast<unsigned>(currentTimeOfDay.seconds().count()),
+            static_cast<unsigned>(currentTimeOfDay.subseconds().count()));
```

Comment:
suggestion: We might want to prefer types from `cstdint` here (e.g. `uint32_t` or `uint64_t` instead of just `unsigned`) and maybe we want to use `aznumeric_cast` too? Not a huge deal though

Reply:
Actually the standard specifies the conversion operators as as [unsigned and int](http://eel.is/c++draft/time#cal), so those are being used explicitly here and not a typedef